azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/config/AzureComputeHttpApiModule.java

```diff
@@ -54,4 +57,15 @@ protected void configure() {
       super.configure();
       bind(OAuthScopes.class).toInstance(OAuthScopes.NoScopes.create());
    }
+   @Provides
+   @Named("subscriptionId")
+   protected String provideSubscriptionId(ProviderMetadata providerMetadata) {
+      String subscriptionid = null;
+      String endpoint = null;
+      endpoint = providerMetadata.getEndpoint();
+      if (endpoint != null) {
```

Comment:
The endpoint can't be null (jclouds enforces that when building the context), so you could remove this conditional and the endpoint variable.